main/src/ca/uwaterloo/flix/language/ast/ParsedAst.scala

```diff
@@ -785,6 +785,16 @@ object ParsedAst {
       */
     case class RestrictableChoose(sp1: SourcePosition, star: Boolean, exp: ParsedAst.Expression, rules: Seq[MatchRule], sp2: SourcePosition) extends ParsedAst.Expression
 
+    /**
+      * ForA Expression.
+      *
+      * @param sp1   the position of the first character in the expression.
+      * @param frags the for-fragments, specifically [[ForFragment.Generator]].
+      * @param exp   the yield-expression.
+      * @param sp2   the position of the last character in the expression.
+      */
+    case class ForA(sp1: SourcePosition, frags: Seq[ForFragment.Generator], exp: ParsedAst.Expression, sp2: SourcePosition) extends ParsedAst.Expression
```

Comment:
What would be a good consistent naming? Brainstorming:

-`ForA` and `ForM`
- `ApplicativeFor` and `MonadicFor`
- `ForAYield` and `ForMYield`

?

Reply:
I think `ApplictiveFor` and `MonadicFor` are better names in the compiler. That way one can easily read off what it means.